File: C++/ReverseArray.cpp
#include <iostream>
#include <vector>
using namespace std;

void reverseArray(vector<int> &arr)
{   
    int n=arr.size();
    for (int i = 0; i < n/2; i++)
    {
    int temp = arr[n - i - 1];
    arr[n - i - 1] = arr[i];
    arr[i] = temp;}
}

int main(int argc, char const *argv[])
{

    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++)
    {   
        cin >> arr[i];
        if(arr[i]==-1) break;
    }

    reverseArray(arr);
    /*for (int i = 0; i < n/2; i++)
    {
        int temp=arr[n-i-1];
        arr[n-i-1]=arr[i];
        arr[i]=temp;
    }*/
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }

    return 0;
}

File: Java/Reverse_Array.java
import java.util.Scanner;

public class Reverse_Array {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        int[] arr = new int[x];
        for (int i = 0; i < x; i++) {
            arr[i] = sc.nextInt();
        }
        int start = 0;
        int end = x - 1;
        while (start < end) {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
        for (int i = 0; i < x; i++) {
            System.out.print(arr[i] + " ");

        }
        sc.close();
    }
}
